### PR TITLE
core#2174: Activity Summary report is missing pagination

### PR DESCRIPTION
**Pull request by monishdeb**
Overview
----------------------------------------
Steps to replicate:

1. Go to Activity Summary report
2. Grouping - Contact
3. Submit

It always shows 50 records without pagination, irrespective of total number of records: Also under stats, there is no indication of actual number of rows. 
 
Before
----------------------------------------
before

After
----------------------------------------
after

Technical Details
----------------------------------------
Issue was ```SELECT FOUND_ROWS()``` was always returning the the default page count - 50, because its acting on last executed query which on temp table. To fix this I have declared a protected variable which track the actual row count and later used to set the form variable ```$_rowCount``` which is not only used in pagination but also used in calculating total number of rows, which was earlier not displayed under stats. 

Comments
----------------------------------------
ping @eileenmcnaughton @seamuslee001

### CODE REVIEW

File: CRM/Report/Form/ActivitySummary.php [hunk 1]
```diff
@@ -20,6 +20,7 @@ class CRM_Report_Form_ActivitySummary extends CRM_Report_Form {
   protected $_phoneField = FALSE;
   protected $_tempTableName;
   protected $_tempDurationSumTableName;
+  protected $_totalRows;
```
Comment (eileenmcnaughton) [inline CRM/Report/Form/ActivitySummary.php:23]: My understanding is the leading underscore is something of a legacy pattern & we have been ditching it over time

File: CRM/Report/Form/ActivitySummary.php [hunk 2]
```diff
@@ -466,7 +467,7 @@ public function buildQuery($applyLimit = TRUE) {
     $this->groupBy(FALSE);
 
     // build the query to calulate duration sum
-    $sql = "SELECT SUM(activity_civireport.duration) as civicrm_activity_duration_total {$this->_from} {$this->_where} {$this->_groupBy} {$this->_having} {$this->_orderBy} {$this->_limit}";
+    $sql = "SELECT SQL_CALC_FOUND_ROWS SUM(activity_civireport.duration) as civicrm_activity_duration_total {$this->_from} {$this->_where} {$this->_groupBy} {$this->_having} {$this->_orderBy} {$this->_limit}";
```
Comment (eileenmcnaughton) [inline CRM/Report/Form/ActivitySummary.php:470]: @monishdeb I think you'll need to do a more deliberate count query - check out https://dev.mysql.com/worklog/task/?id=12615 - this is deprecated in mysql 8

**Comment by monishdeb** [comment]
@eileenmcnaughton I have updated the PR, where renamed the form variable to ```totalRows``` and used a separate query with COUNT(*) as suggested on https://dev.mysql.com/worklog/task/?id=12615.Here's the [commit](https://dev.mysql.com/worklog/task/?id=12615)

**Comment by eileenmcnaughton** [comment]
I was able to replicate this bug and confirm that this PR fixes it

**Pull request merged**